### DISCUSSION TITLE
Reactive UI routing for MAUI

### DISCUSSION THREAD

**Discussion by mStylias** [reactiveui/ReactiveUI#3387] [Q&A]
Hey everyone I am very new to Reactive UI and I've been trying to create a cross platform app that uses the same viewmodels on a MAUI and an Avalonia project. However I can't get the navigation to work in Maui. I've been trying to do it following the instructions of [this video](https://youtu.be/r7ITy7FALTc?list=PLFJQnCcZXWjtQOb-V44yS0FnUHsccS32n). So I wanted to ask
1) Is Reactive UI routing supported on MAUI?
2) Do you know any source that has a demo application with routing implemented on MAUI? (I have been searching the web, but I can only find samples for Xamarin.Forms)

**Comment by mStylias** [comment]
After some research and some demos I found that routing is supported on MAUI. The instructions on the video work with a slight twist. I've made a comment on youtube that explains how. Still if anybody knows any structured course from zero to hero on ReactiveUI it would be very welcome.

**Reply by NickDarvey** [reply]
This might be worth a try:
https://kent-boogaart.com/you-i-and-reactiveui/

**Reply by RAMESHKUMAR502** [reply]
HI Nick,

I have ported Xamarin to .net MAUI .

But i see below issue On MAUI

NavigationPage must have a root Page before being used. Either call PushAsync with a valid Page, or pass a Page to the constructor before usage.

Below is the code i have used

this.MainPage = RxApp.SuspensionHost
.GetAppState()
.CreateMainPage();
